src/zenml/integrations/lightgbm/materializers/lightgbm_dataset_materializer.py

```diff
@@ -47,15 +47,15 @@ def load(self, data_type: Type[Any]) -> lgb.Dataset:
         filepath = os.path.join(self.uri, DEFAULT_FILENAME)
 
         # Create a temporary folder
-        temp_dir = tempfile.mkdtemp(prefix="zenml-temp-")
-        temp_file = os.path.join(str(temp_dir), DEFAULT_FILENAME)
+        with tempfile.TemporaryDirectory(prefix="zenml-temp-") as temp_dir:
+            temp_file = os.path.join(str(temp_dir), DEFAULT_FILENAME)
 
-        # Copy from artifact store to temporary file
-        fileio.copy(filepath, temp_file)
-        matrix = lgb.Dataset(temp_file, free_raw_data=False)
+            # Copy from artifact store to temporary file
+            fileio.copy(filepath, temp_file)
+            matrix = lgb.Dataset(temp_file, free_raw_data=False)
```

Comment:
Ah, this was a tricky bug to catch :) With the current implementation a `matrix` is lazy loaded later on and if you dump it to a temp file with clean-up, it will fail later on. So clean-up here is not feasible without heavy rework.